crates/next-core/src/next_config.rs

```diff
@@ -0,0 +1,216 @@
+use std::collections::HashMap;
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use turbo_tasks::{trace::TraceRawVcs, Value};
+use turbo_tasks_fs::{to_sys_path, FileContent, FileSystemPathVc};
+use turbopack_core::{
+    chunk::ChunkingContextVc, context::AssetContextVc, source_asset::SourceAssetVc,
+};
+use turbopack_ecmascript::{
+    chunk::EcmascriptChunkPlaceablesVc, EcmascriptInputTransformsVc, EcmascriptModuleAssetType,
+    EcmascriptModuleAssetVc,
+};
+use turbopack_node::evaluate::{evaluate, JavaScriptValue};
+
+use crate::embed_js::next_asset;
+
+#[turbo_tasks::value(transparent)]
+pub struct NextConfigValue(NextConfig);
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, TraceRawVcs)]
+#[serde(rename_all = "camelCase")]
+pub struct NextConfig {
+    pub config_file: Option<String>,
+    pub config_file_name: String,
+    pub typescript: Option<TypeScriptConfig>,
+    pub react_strict_mode: Option<bool>,
+    pub experimental: Option<ExperimentalConfig>,
+    pub env: Option<HashMap<String, String>>,
+    pub compiler: Option<CompilerConfig>,
+    pub images: ImageConfig,
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "camelCase")]
+pub struct TypeScriptConfig {
+    pub ignore_build_errors: Option<bool>,
+    pub ts_config_path: Option<String>,
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "camelCase")]
+pub struct ImageConfig {
+    pub device_sizes: Vec<u16>,
+    pub image_sizes: Vec<u16>,
+    pub path: String,
+    pub loader: ImageLoader,
+    pub domains: Vec<String>,
+    pub disable_static_images: bool,
+    #[serde(rename(deserialize = "minimumCacheTTL"))]
+    pub minimum_cache_ttl: u32,
+    pub formats: Vec<ImageFormat>,
+    #[serde(rename(deserialize = "dangerouslyAllowSVG"))]
+    pub dangerously_allow_svg: bool,
+    pub content_security_policy: String,
+    pub remote_patterns: Vec<RemotePattern>,
+    pub unoptimized: bool,
+}
+
+impl Default for ImageConfig {
+    fn default() -> Self {
+        Self {
```

Comment:
Nit: I should have done this myself when adding the old default, but let's add a link:
```suggestion
        // https://github.com/vercel/next.js/blob/327634eb/packages/next/shared/lib/image-config.ts#L100-L114
        Self {
```